test/Makefile

```diff
@@ -12,10 +12,33 @@ $(TDIRS):
 clean:
 	-rm -fr out
 
-update:
+update: out/cmpll/ll_curr.txt out/cmpll/ll_prev.txt
+	comm -13 out/cmpll/ll_curr.txt out/cmpll/ll_prev.txt | xargs rm -fv
+	rm out/cmpll/ll_curr.txt out/cmpll/ll_prev.txt
 	for f in out/*/ll/*.ll; do \
 		ll=`echo $$f | sed -e 's;/ll/;/;' -e 's;^out;ll;'`; \
 		cmp -s $$f $$ll || (mkdir -p `dirname $$ll` && cp -p $$f $$ll); \
 	done
 
-.PHONY: $(TARGETS) $(TDIRS) clean update
+diff: out/cmpll/ll_curr.txt out/cmpll/ll_prev.txt
+	@diff out/cmpll/ll_curr.txt out/cmpll/ll_prev.txt | sed -e 's;^< ll/\(.*\)/;< out/\1/ll/;' -e '/^[<>]/!d'
```

Comment:
Can you explain what this is doing?  It's a lot more complicated that I expected.

Reply:
I wanted to produce an output like :
```
> ll/01-ifelse/2-v.ll
< out/01-int/ll/add3-v.ll
out/cmpll/01-function/1-v.diff
```
Here `ll/01-ifelse/2-v.ll` is to be removed, `out/01-int/ll/add3-v.ll` is a new file to be added.

I produce this by listing the two sets of files, massaging it to have the same format and getting the diff between them. Same has been used in `update:` to remove files. Once I get the diff I undo the massaging to get the original file names (so you may click on it to view).